topic04-git-introduction/book-git-tutorial/01.Basics.md

```diff
@@ -0,0 +1,30 @@
+# **_Git guide to commands_**
+
+This manual contains necessary commands for git when working on your own, contributing to other projects and working as a team.
+
+###### What is git about?
+
+Git is a network that allows you to manage your code easily while working as a team or contributing to other repos, It allows you to merge code without having to copy and paste it from another file or files. Git also is able to host your website but it does not give you a proper full domain. 
+
+You are also able to use it to fork a repo which creates a copy of a repo in your github account and allows you to do pull requests which allow you to basically create requests that allow you to suggest changes to the original owner of the repo which they can review, accept or decline and if they accept they can merge it into their original repo.
+
+#### **_The basics:_**
+`Git push`
+`Git pull`
+`Git add (filename)`
+`Git status`
+`Git commit -m “first commit”` 
+
+- These commands are used to add files to your repository through a terminal or git bash.
+- `git pull`, fetches changes from the remote master and merges changes, it can also be used like this `git pull (remote) (branchname)`.
+- To add files from your machine through a terminal or git bash we first use the command `git add <filename>`. 
+                **Git add basically allows the terminal or git bash to grab your       
+                file off your machine and allows it to be committed and   
+                pushed to a github repo.**
+
+- Then we use `git commit -m “insert your own message based on the commit”` but before you commit always check if the files are staged using `git status`. To commit the files to a repo after that we check if it got staged or not by using git status to check if it worked it’s green and it’s staged and ready to push, if it’s red it’s unstaged and `git commit` stage’s it. 
+
+After we committed and it shows the filename in green when using git status we use git push to push the file into the repo. 

```

Comment:
Once you commit, there will be no files in green as you have already committed them. I suggest adding `git log` here to check and make sure that you have successfully committed before pushing your changes to the remote repo.